### ISSUE TITLE
How to reference the struct itself in the struct ?

### ISSUE THREAD

**Issue by Vexcited** [Koromix/koffi#101]
Hi! I have a C struct looking like this :
```h
struct something {
  struct something *next;
  char *name;
  char *description;
};
```

But when I try to do something like :
```javascript
koffi.struct('something', {
  next: 'something *',
  name: 'char *',
  description: 'char *'
});
```

I get an error `TypeError: Unknown or invalid type name 'something *'`

Is there any way to get around this ? I'm pretty new to FFI stuff and sadly the lack of documentation/examples (thankfully Koffi's website as a lot of examples to get started) makes the task a bit more difficult for me :/

**Comment by Vexcited** [comment]
Found out using...

```javascript
koffi.struct('something', {
  next: 'void *',
  name: 'char *',
  description: 'char *'
});
```

worked somehow, but I'll keep this issue open if there's any better suggestion.

**Comment by Koromix** [comment]
I've changed the Koffi to allow self-referentials structs.
I've just published Koffi 2.6.5: https://koffi.dev/changelog#koffi-2-6-5-2023-10-28

**Comment by Koromix** [comment]
This is possible in Koffi 2.10 and later versions, which allow you to redefine an opaque type.

Documentation here: https://koffi.dev/misc#circular-references

```js
const koffi = require("koffi");

// Make it opaque to use it in the definition of B
const A = koffi.opaque("A");

const B = koffi.struct("B", {
    a: "A*"
});

// Finalize A type
koffi.struct(A, {
    b: "B*"
});
```
